crates/volta-core/src/run/parser.rs

```diff
@@ -345,12 +345,15 @@ pub struct LinkArgs<'a> {
 }
 
 impl<'a> LinkArgs<'a> {
-    pub fn executor(self, platform: Platform) -> Fallible<Executor> {
+    pub fn executor(self, platform: Platform, project_name: Option<String>) -> Fallible<Executor> {
         if self.tools.is_empty() {
             // If no tools are specified, then this is a bare link command, linking the current
             // project as a global package. We treat this exactly like a global install
```

Comment:
Does this comment still actually hold without modification? Also, can we add to it slightly to explain what we're doing with the `name` handling? At first read, it wasn't obvious which piece of the new code it meant, exactly (because it refers to the old code!) and it *doesn't* explain the `name` handling, so I was left slightly scratching my head.

Reply:
That's a good call out, I'll update it to reflect that we treat it like a global install that we definitely know the name up-front (global installs can be from an arbitrarily named tarball or git link, so we don't always know the name a priori).